tests/testthat/test-generate_triangle.R

```diff
@@ -199,6 +199,90 @@ test_that("generate_triangle handles custom structure with vector", {
   expect_identical(result_custom, expected_custom)
 })
 
+test_that("generate_triangle can generate something with all NAs at end", {
+  exp_result <- matrix(
+    c(
+      1, 3, 5, 7, 9,
+      4, 5, 9, 4, 3,
+      1, 6, 4, NA, NA,
+      3, NA, NA, NA, NA
+    ),
+    nrow = 4,
+    byrow = TRUE
+  )
+
+  trunc_rt <- matrix(
+    c(
+      1, 3, 5, 7, 9,
+      4, 5, 9, 4, 3,
+      1, 6, 4, 4, 3,
+      3, 8, 4, 6, 1
+    ),
+    nrow = 4,
+    byrow = TRUE
+  )
+  actual_result <- generate_triangle(trunc_rt,
+    structure = c(1, 2)
+  )
+  expect_identical(exp_result, actual_result)
+})
+
+test_that("generate_triangle can handle a structure where first row has more than one NA", { # nolint
+  exp_result <- matrix(
+    c(
+      1, 3, 5, 7, 9,
+      4, 5, 9, 4, 3,
+      1, 6, 4, NA, NA,
+      3, 8, NA, NA, NA
+    ),
+    nrow = 4,
+    byrow = TRUE
+  )
+
+  trunc_rt <- matrix(
+    c(
+      1, 3, 5, 7, 9,
+      4, 5, 9, 4, 3,
+      1, 6, 4, 4, 3,
+      3, 8, 4, 6, 1
+    ),
+    nrow = 4,
+    byrow = TRUE
+  )
+  actual_result <- generate_triangle(trunc_rt,
+    structure = c(2, 1)
+  )
+  expect_identical(exp_result, actual_result)
+})
+
+test_that("generate_triangle can handle a structure ending with 2 NAs", {
```

Comment:
@seabbs I am still not following, it would be helpful if you could just suggest the test you want. 

This test is testing the edge case similar to the problem that occurred which is that the structure must allow for more than one NA at the end (here it is 2, I can make it more if we want).

Reply:
The issue here is that in all of these the implied last report has 0 NAs as you are specifying the reporting for line 262. We want to check that the code can handle settings where the line for 261 is not zero or 1 NA but some other number as that is the implied and not explicit structure. That is currently missing from the tests. As this is a sticking point I think its fine to write a new issue to address this. I don't have time to work on this for the next two weeks.